azurerm/internal/services/eventgrid/tests/eventgrid_system_topic_resource_test.go

```diff
@@ -0,0 +1,222 @@
+
+		resp, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: EventGrid System Topic %q (resource group: %s) does not exist", name, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on EventGridSystemTopicsClient: %s", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMEventGridSystemTopic_basic(data acceptance.TestData) string {
+	// TODO: confirm if this is still the case
+	// currently only supported in "West Central US" & "West US 2"
+	location := "westus2"
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestegst%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_eventgrid_system_topic" "test" {
+  name                = "acctestegst%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  source              = azurerm_storage_account.test.id
+  topic_type          = "Microsoft.Storage.StorageAccounts"
+}
+`, data.RandomInteger, location, data.RandomIntOfLength(12), data.RandomIntOfLength(10))
+}
+
+func testAccAzureRMEventGridSystemTopic_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMEventGridSystemTopic_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_eventgrid_system_topic" "import" {
+  name                = azurerm_eventgrid_system_topic.test.name
+  location            = azurerm_eventgrid_system_topic.test.location
+  resource_group_name = azurerm_eventgrid_system_topic.test.resource_group_name
+  source              = azurerm_eventgrid_system_topic.test.source
+  topic_type          = azurerm_eventgrid_system_topic.test.topic_type
+}
+`, template)
+}
+
+func testAccAzureRMEventGridSystemTopic_basicWithTags(data acceptance.TestData) string {
+	location := "westus2"
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestegst%d"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_eventgrid_system_topic" "test" {
+  name                = "acctestegst%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  source     = azurerm_storage_account.test.id
+  topic_type = "Microsoft.Storage.StorageAccounts"
+
+  tags = {
+    "foo" = "bar"
```

Comment:
could we ensure capitals work here
```suggestion
    "Foo" = "Bar"
```

Reply:
done